cpp/oneapi/dal/backend/primitives/sort/sort.hpp

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+template <typename Float>
+struct float2uint_map;
+
+template <>
+struct float2uint_map<float> {
+    using integer_t = std::uint32_t;
+};
+
+template <>
+struct float2uint_map<double> {
+    using integer_t = std::uint64_t;
+};
+
+/// @tparam Float Floating-point type used for storing input values
+/// @tparam Index Integer type used for storing input indices
+template <typename Float, typename Index = std::uint32_t>
+class radix_sort_indices_inplace {
+    using radix_integer_t = typename float2uint_map<Float>::integer_t;
```

Comment:
If `float2uint_map` is used only once, I'd suggest to remove this type and use standard type traits:

```cpp
static_assert(std::is_same_v<float, Float> || std::is_same_v<double, Float>)
using radix_integer_t = std::conditional_t<std::is_same_v<float, Float>, std::uint32_t, std::uint64_t>;
```

Reply:
I'd support @makart19 solution, but add `static_assert` to `float2uint_map`

Reply:
added static_assert